lib/src/values/logic_value.dart

```diff
@@ -115,16 +118,56 @@ abstract class LogicValue {
   /// Constructs a [LogicValue] from [it].
   ///
   /// The order of the created [LogicValue] will be such that the `i`th entry in
-  /// [it] corresponds to the `i`th bit.  That is, the 0th element of [it] will
-  /// be the 0th bit of the returned [LogicValue].
+  /// [it] corresponds to the `i`th group of bits.  That is, the 0th element of
+  /// [it] will be the least significant chunk of bits of the returned
+  /// [LogicValue].  Bits within each element of [it] are kept in the same
+  /// order as they were originally.
   ///
+  /// For example:
   /// ```dart
-  /// var it = [LogicValue.zero, LogicValue.x, LogicValue.one];
+  /// var it = [LogicValue.zero, LogicValue.x, LogicValue.ofString('01xz')];
   /// var lv = LogicValue.of(it);
-  /// print(lv); // This prints `3b'1x0`
+  /// print(lv); // This prints `6'b01xzx0`
   /// ```
-  static LogicValue of(Iterable<LogicValue> it) => LogicValue.ofString(
-      it.map((e) => e.toString(includeWidth: false)).toList().reversed.join());
+  ///
+  static LogicValue of(Iterable<LogicValue> it) => it.fold(LogicValue.empty,
+      (previousValue, element) => element._concatenate(previousValue));
```

Comment:
I like the idea of this much better than concatenating strings and parsing, but I'm sure we can implement this better.  Doing sequential concatenation will create n-2 temporary values and will potentially take O(n^2) time.  Consider concatenating 1000 bits; the first bit will be shifted left by 1 999 times.  
Instead, I recommend a 2-pass solution.  In the first pass, you're going to sum the widths and check if all inputs are FIlledLogicValues.  Then you can create the final output kind of LogicValue and shift each of the _values into the right place in that output.

Reply:
Thank you for the help and suggestions (here and outside of this thread)!  The implementation I just pushed based on your feedback is ~3x faster than the one this comment referred to originally.